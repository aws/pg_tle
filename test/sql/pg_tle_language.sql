/*
*
* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
*/

\pset pager off

CREATE EXTENSION pg_tle;

-- attempt to create a TLE using a C-based function. This will error on
-- CREATE EXTENSION
SELECT pgtle.install_extension(
  'tle_c',
  '1.0',
  FALSE,
  'this will fail',
$_pgtle_$
  CREATE FUNCTION geo_distance (point, point)
  RETURNS float8
  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE AS '$libdir/earthdistance';
$_pgtle_$
);

-- this will fail
CREATE EXTENSION tle_c;

-- cleanup
SELECT pgtle.uninstall_extension('tle_c');
DROP EXTENSION pg_tle;
DROP SCHEMA pgtle;
DROP ROLE pgtle_staff;
DROP ROLE pgtle_admin;
