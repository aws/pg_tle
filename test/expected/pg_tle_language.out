/*
*
* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
*/
\pset pager off
CREATE EXTENSION pg_tle;
-- attempt to create a TLE using a C-based function. This will error on
-- CREATE EXTENSION
SELECT pgtle.install_extension(
  'tle_c',
  '1.0',
  FALSE,
  'this will fail',
$_pgtle_$
  CREATE FUNCTION geo_distance (point, point)
  RETURNS float8
  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE AS '$libdir/earthdistance';
$_pgtle_$
);
 install_extension 
-------------------
 t
(1 row)

-- this will fail
CREATE EXTENSION tle_c;
ERROR:  language "c" is not trusted and cannot be used in a trusted-language function.
-- cleanup
SELECT pgtle.uninstall_extension('tle_c');
 uninstall_extension 
---------------------
 t
(1 row)

DROP EXTENSION pg_tle;
DROP SCHEMA pgtle;
DROP ROLE pgtle_staff;
DROP ROLE pgtle_admin;
